src/select/__tests__/select.test.tsx

```diff
@@ -450,6 +451,162 @@ describe.each([false, true])('expandToViewport=%s', expandToViewport => {
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.findDisabledReason()).toBe(null);
+      });
+
+      test('has no tooltip without disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+        wrapper.findTrigger()!.keydown(KeyCode.down);
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.findDisabledReason()).toBe(null);
+      });
+
+      test('open tooltip when the item is highlighted', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+                disabledReason: 'disabled reason',
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+        wrapper.findTrigger().keydown(KeyCode.down);
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.findDisabledReason()!.getElement()).toHaveTextContent(
+          'disabled reason'
+        );
+      });
+
+      test('has no aria-describedby by default', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions,
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has no aria-describedby without disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has aria-describedby with disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+                disabledReason: 'disabled reason',
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.getElement()).toHaveAttribute('aria-describedby');
+      });
+
+      test('has hidden element (linked to aria-describedby) with disabledReason', () => {
```

Comment:
Can this be merged with another test rendering the same props configuration?

Reply:
Technically yes, but I followed "one test one assertion" pattern to keep things easier to debug if something goes wrong

Reply:
Not a strong opinion here, but I think deduplicating tests actually helps than reviewing multiple failed tests with the same setup config